homeassistant/components/uptimerobot/__init__.py

```diff
@@ -53,3 +42,62 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class UptimeRobotDataUpdateCoordinator(DataUpdateCoordinator):
+    """Data update coordinator for Uptime Robot."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry_id: str,
+        dev_reg: DeviceRegistry,
+        api: UptimeRobot,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_method=self._async_update_data,
+            update_interval=COORDINATOR_UPDATE_INTERVAL,
+        )
+        self._config_entry_id = config_entry_id
+        self._device_registry = dev_reg
+        self._api = api
+
+    async def _async_update_data(self) -> list[UptimeRobotMonitor] | None:
+        """Update data."""
+        try:
+            response = await self._api.async_get_monitors()
+        except UptimeRobotException as exception:
+            raise UpdateFailed(exception) from exception
+        else:
+            if response.status != API_ATTR_OK:
+                raise UpdateFailed(response.error.message)
+
+        monitors: list[UptimeRobotMonitor] = response.data
+
+        current_monitors = {
+            list(device.identifiers)[0][1]
+            for device in self._device_registry.devices.values()
+            if self._config_entry_id in device.config_entries
+            and list(device.identifiers)[0][0] == DOMAIN
```

Comment:
This check doesn't look correct. We can't be sure that the first identifier belongs to this integration if we're guarding against a device with more then one connected integration (domain). We're also not iterating all the identifiers in this set comprehension.

If there's no risk for multiple identifiers from different integrations, it would be enough to check for matching config entry id. For this we have this helper:
https://github.com/home-assistant/core/blob/7d29eb282bf1670692d6bc4d38d76dec22029279/homeassistant/helpers/device_registry.py#L652

Reply:
RIGHT!
The integration allows multiple entries, this will mess with other entries...
Will do a followup later today

Reply:
https://github.com/home-assistant/core/pull/54256